File: python/basic/algorithm/company/zigzeg/ExtractOddNumbers.js
function extractOddNumbers(n) {
    let result = n.match(/\d*[13579]+/g);
    result.forEach(function (value, index) {
        if (value % 2 !== 0) {
            result[index] = Math.pow(value, 2)
        }
    })


    let sum = result.reduce(function (a, b) {
        return a + b;
    }, 0);
    return console.log(sum)
}

extractOddNumbers("ab2v9bc13j5jf4jv21")

File: python/basic/algorithm/company/zigzeg/ExtractOddNumbers.py
import re


def get_number(n):
    result = list(filter(lambda x: x % 2 != 0, map(int, re.findall('\d+', n))))
    return sum(pow(x, 2) for x in result)


print(get_number("ab2v9bc13j5jf4jv21"))
